### PR TITLE
cmd/publish-artifacts: properly seek to start of file

### PR DESCRIPTION
**Pull request by tamird**
Fixes an issue where the Windows "latest" archive was empty because the
binary file wasn't being correctly rewound after the versioned archived
was written on Windows.

### CODE REVIEW

**Comment by tamird** [comment]
We might want to cherry pick this into 1.0 @mberhault.

**Review by jordanlewis — APPROVED** [review]
LGTM as long as you tested the output binary

**Comment by tamird** [comment]
Urgh, I just tried to test it, but looks like @benesch's https://github.com/cockroachdb/cockroach/pull/15719 or other recent change broke Windows compilation generally:
```
/go/native/x86_64-w64-mingw32/rocksdb/librocksdb.a(malloc_stats.cc.obj):malloc_stats.cc:(.text+0xcc): undefined reference to `je_malloc_stats_print'
collect2: error: ld returned 1 exit status
Makefile:127: recipe for target 'buildoss' failed
make: *** [buildoss] Error 2
2017/05/06 23:35:12 [make buildoss TYPE=release-windows GOFLAGS= SUFFIX= TAGS=]: exit status 2
exit status 1
```

I'll wait until that's fixed and test & merge then.

**Comment by benesch** [comment]
Yup, sorry. Really not clear why to my subconscious, but I haven't been on
a computer to investigate.

Feel free to dig in if you're bored and have time to kill. #15719 was small
so shouldn't be too hard to track down.

**Comment by tamird** [comment]
Found it, and it's absurd: https://github.com/cockroachdb/cockroach/pull/15744.

**Comment by tamird** [comment]
OK, with #15744 I was able to test and confirm that the zip files are produced correctly. I'll merge this after #15744.

**Pull request merged**
